sdk/core/System.ClientModel/src/Pipeline/HttpClientPipelineTransport.cs

```diff
@@ -0,0 +1,212 @@
+
+    private bool _disposed;
+
+    public HttpClientPipelineTransport() : this(CreateDefaultClient())
+    {
+        // We will dispose the httpClient.
+        _ownsClient = true;
+    }
+
+    public HttpClientPipelineTransport(HttpClient client)
+    {
+        Argument.AssertNotNull(client, nameof(client));
+
+        _httpClient = client;
+
+        // The caller will dispose the httpClient.
+        _ownsClient = false;
+    }
+
+    private static HttpClient CreateDefaultClient()
+    {
+        // The following settings are added in Azure.Core and are not included
+        // in System.ClientModel.  If needed, we will migrate them to ClientModel
+        // over time.
+        //   - SSL settings
+        //   - Proxy settings
+        //   - Cookies
+        //   - MaxConnectionsPerServer
+        //   - PooledConnectionLifetime
+
+        HttpClientHandler handler = new HttpClientHandler()
+        {
+            AllowAutoRedirect = false
+        };
+
+        return new HttpClient(handler)
+        {
+            // Timeouts are handled by the pipeline
+            Timeout = Timeout.InfiniteTimeSpan,
+        };
+    }
+
+    protected override PipelineMessage CreateMessageCore()
+    {
+        PipelineRequest request = new HttpPipelineRequest();
+        PipelineMessage message = new PipelineMessage(request);
+
+        return message;
+    }
+
+    protected sealed override void ProcessCore(PipelineMessage message)
+    {
+#pragma warning disable AZC0102 // Do not use GetAwaiter().GetResult().
+
+#if NET6_0_OR_GREATER
+
+        ProcessSyncOrAsync(message, async: false).GetAwaiter().GetResult();
+
+#else
+
+        // We do sync-over-async on netstandard2.0 target.
+        // This can cause deadlocks in applications when the threadpool gets saturated.
+        // The resolution is for a customer to upgrade to a net6.0+ target,
+        // where we are able to provide a code path that calls HttpClient native sync APIs.
+
+        ProcessSyncOrAsync(message, async: true).AsTask().GetAwaiter().GetResult();
+
+#endif
+
+#pragma warning restore AZC0102 // Do not use GetAwaiter().GetResult().
+    }
+
+    protected sealed override async ValueTask ProcessCoreAsync(PipelineMessage message)
+        => await ProcessSyncOrAsync(message, async: true).ConfigureAwait(false);
+
+    private async ValueTask ProcessSyncOrAsync(PipelineMessage message, bool async)
+    {
+        if (message.Request is not PipelineRequest request)
+        {
+            throw new InvalidOperationException($"The request type is not compatible with the transport: '{message.Request?.GetType()}'.");
+        }
+
+        using HttpRequestMessage httpRequest = HttpPipelineRequest.BuildHttpRequestMessage(request, message.CancellationToken);
+
+        OnSendingRequest(message, httpRequest);
+
+        HttpResponseMessage responseMessage;
+        Stream? contentStream = null;
+        message.Response = null;
+
+        try
+        {
+#if NET5_0_OR_GREATER
```

Comment:
I thought we wanted to multitarget to NET6 and the rest. This creates another slice at NET5.

Reply:
@jsquire had wanted to keep the TFMs the same even when they were referencing unsupported platforms when I discussed it with him last.  @jsquire do you care about this one and want to share your argument?  I'm more in favor of only using TFMs for platforms we support, but I don't feel strongly one way or another so happy to do whatever the group decides is best here.

Reply:
I checked in with @jsquire on this and he doesn't feel strongly, so will fix per @KrzysztofCwalina's request.